sycl/include/CL/__spirv/spirv_vars.hpp

```diff
@@ -24,13 +24,17 @@ extern "C" const __constant size_t_vec __spirv_BuiltInGlobalOffset;
   template <> size_t get##POSTFIX<1>() { return __spirv_BuiltIn##POSTFIX.y; }  \
   template <> size_t get##POSTFIX<2>() { return __spirv_BuiltIn##POSTFIX.z; }
 
+namespace __spirv {
```

Comment:
What is the reason of

Reply:
Good question.
Originally we re-used OpenCL C++ approach including naming requirements (SPIR-V built-ins must been declared in `__spirv` namespace).
We refactored our code and now there are no uses of `__spirv` namespace in our code base.

I think eventually we should remove `__spirv` directory with the header files and re-use declarations of SPIR-V opcodes from SPIRV-Headers repository directly.
The reasonable place for utility functions used for SYCL is `sycl::details`.

It's better to be done in a separate PR.

Reply:
Yes, not a huge fan of at least the DEFINE_INIT_SIZES being wrapped into the `__spirv` namespace as they were previously in the detail namespace, which was a more backend agnostic namespace. But I can live with it for now as long as we're aiming to make it a little less SPIRV centric in the future! Thanks for the great work.

Reply:
Here is my justification for putting this into __spirv:
1) This stuff is not really SYCL-specific, it is really spirv-specific, so I think sycl::details is not the right place for it.
2) On the other hand, we need some namespace not to pollute the top-level scope.

Hence __spirv, to match __spirv_<name> style for SPIRV built-ins.

Reply:
My issue will circle back around to the original discussion in the original pull request, unfortunately. So it's probably not worth wasting your time too much as I'm fine with this patch! 

I agree partly with your justification as these functions are currently named related to SPIRV built-ins (even though the Init calls really don't need to be as far as I'm aware), I just think it's unfortunate that the SPIRV namespace is wrapped around these Init functions as they are still the top level function invoked to retrieve all of the built-in information for the device. As they now have to be prefixed with __spirv:: rather than the previously agnostic detail::. Mainly as I have to stick our SPIR InitX function into a slightly confusing SPIRV namespace or wrap an #ifdef around every invocation of these Init calls. Not too big a deal but I would enjoy a degree of separation. But yeah, this is related to the initial discussion and a battle for another pull request as we agreed upon previously.

Reply:
@agozillon - I see your point. detail would make it look device agnostic indeed. There will still be other similar stuff in __spv namespace though. In another thread we I think agreed to consider introducing a layer abstracting away device-side builtins and/or runtime. As you suggest, we can do it in another pull request and resolve both __spv and __spirv. ?

Reply:
Yeah, that's fine by me.